contributors/design-proposals/service-injection-policy.md

```diff
@@ -0,0 +1,274 @@
+# Service Injection Policy
+
+## Abstract
+
+Describes a policy resource that allows loose coupling from services to the pods
+that consume them.
+
+## Motivation
+
+Consuming a service involves more than just connectivity.  In addition to
+coordinates to reach the service, credentials and non-secret configuration
+parameters are typically needed to use the service.  The primitives for this
+already exist, but a gap exists where loose coupling is desired: it should be
+possible to inject pods with the information they need to use a service on a
+service-by-service basis, without the pod authors having to incorporate the
+information into every pod spec where it is needed.
+
+## Constraints and Assumptions
+
+1.  New mechanisms must be made to work with controllers such as deployments and
+    replicasets that create pods
```

Comment:
I think some additional qualification is needed here, and what I had had in mind is that existing controllers that create generations of pods should be made to recreate pods when a new SIP is created that would alter them.